backend/urls.py

```diff
@@ -1,13 +1,14 @@
 from django.urls import include, path
-from .views import ListPosts
+from .views import ListPosts, ListComments
 from rest_framework import routers 
-from .api import PostViewSet
+from .api import PostViewSet, CommentViewSet
 
 # Wire up our API using automatic URL routing.
 # Additionally, we include login URLs for the browsable API.
 
 router = routers.DefaultRouter()
 router.register('posts', PostViewSet, 'posts')
+router.register('posts', CommentViewSet, 'comments')
```

Comment:
why is the prefix posts for comments? this will override the postviewset urls

Reply:
I tested backend/post and backend/posts/id

It didn't break

Reply:
You are registering a viewset that has an @action decorator with comments and a pk, which is why posts/id/comments is working. your viewset only exists to have that action decorator for comments, so it should go in the postviewset. Registering commentviewset with the same prefix just to contain a decorator seems very convoluted. You should read up on the documentation here: https://www.django-rest-framework.org/api-guide/routers/#routing-for-extra-actions